packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -115,19 +152,62 @@ export const makeEditGenerator = (): AsyncGenerator<Operation, FuzzTestState> =>
 		};
 	}
 
+	async function transactionStartGenerator(state: EditState): Promise<TransactionStartOp> {
+		return {
+			fuzzType: "transactionStart",
+		};
+	}
+
+	async function transactionCommitGenerator(state: EditState): Promise<TransactionCommitOp> {
+		return {
+			fuzzType: "transactionCommit",
+		};
+	}
+
+	async function transactionAbortGenerator(state: EditState): Promise<TransactionAbortOp> {
+		return {
+			fuzzType: "transactionAbort",
+		};
+	}
+
 	const baseEditGenerator = createWeightedAsyncGenerator<FuzzChange, EditState>([
-		[insertGenerator, 5],
+		[
+			insertGenerator,
+			opWeights.insert,
+			({ testTreeProvider, treeIndex }) =>
+				transactionsInProgress(testTreeProvider.trees[treeIndex]),
```

Comment:
We don't need to check whether a transaction is in progress. If no transaction is in progress, and we attempt such an edit, then SharedTree will automatically wrap the edit in a transaction. (Same for delete, and setPayload).
We do however need to check when calling abort or commit, which I see you are doing.

Reply:
removed unnecessary generator conditions